Build airsonic from source and allow patches

ToxicFrog opened the pull request:
###### Motivation for this change
The current package simply installs the Airsonic binary. This change (a) makes it build from source instead, so that patches can be applied, and (b) adds a patch that inverts the (hard-coded) podcast sort order so that "play all" on a podcast plays them in chronological order rather than reverse chronological.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ToxicFrog committed:
airsonic: switch to building from source code rather than installing binary

This makes it possible to apply patches to it at build time.

ToxicFrog committed:
Add myself to maintainer list

delroth commented:
+1 to building from source. However I think that patch would be a better fit for your own nixpkgs overlay, or contributed upstream directly.

ToxicFrog commented:
Ok, it looks like you can't use mavenix in nixpkgs, or at least you have to use it in a completely different fashion -- pre-download it and include it in the .nix files for that package?

ToxicFrog commented:
Patch excised and moved to my personal overlays while I work on the additional infrastructure needed to get it upstream-ready.
Airsonic build updated to 10.6.2 to match nixpkgs master, and rebased.

ToxicFrog commented:
Rebased against unstable to resolve merge conflicts in maintainers file.
